enhancements/kueue/openshift-kueue.md

```diff
@@ -0,0 +1,727 @@
+During the installation of a cluster or installing of critical operators, it is necessary to not have Kueue intercept these workloads. 
+We do not want kueue gating GPUOperator Pods or any other components like that.
+
+One avenue we have to make this safer is to allow for namespace opt-in. 
+This will be necessary for pod integration and we are considering this for Job based integration also.
+
+Kueue has concepts of managedJobsWithoutQueueName and managedJobsWithNamespaceSelector. 
+These can add a default name for all workloads or target kueue on specific namespaces. managedJobsWithNamespaceSelector relies on a particular label for the namespace. 
+Admins would be required to add a label for all kueue managed namespaces.
+
+For tech preview, we will not enforce Kueue quotas on user workloads unless there is a kueue-managed label on the namespace.
+
+
+#### Telemetry
+
+Kueue has many useful metrics for adminstrators. We will use openshift-monitoring so that one can view metrics
+via the metrics dashboard in OCP console.
+
+The common requests from Product for telemetry are the following:
+
+a. Who is installing Kueue from OperatorHub (Handled by OLM)
+
+b. What kind of resources are customers using in their Cluster Queues
+  Kueue has a metric called `kueue_cluster_queue_resource_usage` which will list the resources.
+
+c. The kind of frameworks Kueue is configured with. 
+We have created an [upstream issue](https://github.com/kubernetes-sigs/kueue/issues/4336) to get this as a kueue metric.
+
+d. AI usage versus Non AI usage.
+
+This is still up in the air but one idea is that we can use telemetry for GPU or RHOAI metrics to determine AI users versus non AI users.
+
+#### Existing Usage of Kueue
+
+RHOAI is a major user of Kueue and they have already released Kueue as GA. 
+The integrations that are GA for RHOAI, as of writing this enhancement, are Kubeflow Training Operator and Ray.
+
+RHOAI also uses a single release of Kueue across the latest 4 releases of Openshift (ie 4.14, 4.15, 4.16, 4.17).
+A request from them for this operator is supporting Kueue across multiple releases of Openshift.
+
+#### Feature gates
+
+Moved to open questions.
+
+### Risks and Mitigations
+
+Kueue is a very fast moving community. They release at least 6 times a year.
+APIs are all in beta and there is some movement to graduate them. 
+But the project does not have a LTS option yet.
+
+To mitigate risk, we are engaging with upstream to define release policies and aim to graduate their critical APIs.
+
+### Drawbacks
+
+Not relevant here.
+
+## Open Questions [optional]
+
+### Autoscaling Future
+
+Autoscaling will be a followup enhancement. 
+Kueue does not provide much guidance on the safe enablement of autoscaling.
+We know that we need a secure way of enabling autoscaler and we should think through that in more detail.
+
+### RHOAI and Kueue Integration
+
+RHOAI is using Kueue as a GA product. We still need to figure out the path with RHOAI and OCP Kueue.
```

Comment:
Using Kueue, what does that mean? How exposed is Kueue to their customers? What are our existing support responsibilities as a company for Kueue?

Reply:
updated this section. PTAL.